packages/client/src/components/TransferToken/components/ButtonGroup.js

```diff
@@ -22,8 +24,12 @@ const ButtonGroup = () => {
     recipientValid,
     selectedNFT,
     coinType,
+    transactionType,
   } = sendModalState;
 
+  const safeAddress =
+    transactionType === TRANSACTION_TYPE.SEND ? address : recipient;
```

Comment:
Address in this case is the sender address, so in case of sending tokens - it's a treasury address, and in case of depositing -> it's the user's address. I will rename this to sender, so it will make more sense!